### PR TITLE
Dev feature submission details graphql

### PR DESCRIPTION
**Pull request by manu-govind**
Graphql changes for submission projection

### CODE REVIEW

File: pstatus-graphql-ktor/src/main/kotlin/gov/cdc/ocio/processingstatusapi/loaders/ReportLoader.kt [hunk 1]
```diff
@@ -99,4 +103,44 @@ class ReportLoader: CosmosLoader() {
...
+    /**
+     *  Get uploadDetails
+     *  @param reports MutableList<Report>
+     *  @return UploadDetails
+     */
+
+    private fun getUploadDetails(reports:MutableList<Report>):UploadDetails {
+
+        // Determine rollup status
+        val rollupStatus = when {
```
Comment (mkrystof) [inline pstatus-graphql-ktor/src/main/kotlin/gov/cdc/ocio/processingstatusapi/loaders/ReportLoader.kt:116]: Please make rollup status an enum.

File: pstatus-graphql-ktor/src/main/kotlin/gov/cdc/ocio/processingstatusapi/models/submission/Properties.kt [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+ * Properties within a issue.
+ *
+ * @property level String?
+ * @property message Int?
+ */
+@GraphQLDescription("Contains issue properties.")
+data class Properties(
+
+    @GraphQLDescription("Issue level")
+    var level : String? = null,
```
Comment (mkrystof) [inline pstatus-graphql-ktor/src/main/kotlin/gov/cdc/ocio/processingstatusapi/models/submission/Properties.kt:15]: Please make level an enum: [ERROR, WARNING]

File: pstatus-graphql-ktor/src/main/kotlin/gov/cdc/ocio/processingstatusapi/models/submission/Provenance.kt [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    var messageId : String? = null,
+
+    @GraphQLDescription("Unique Identifier for that message")
+    var messageUUID : String? = null,
+
+    @GraphQLDescription("MessageHash value")
+    var messageHash: String? = null,
+
+    @GraphQLDescription("Single or Batch message")
+    var singleOrBatch: String? = null,
```
Comment (mkrystof) [inline pstatus-graphql-ktor/src/main/kotlin/gov/cdc/ocio/processingstatusapi/models/submission/Provenance.kt:25]: This should be `aggregation` not `singleOrBatch`.  Also, please make an enum: [SINGLE, BATCH].

File: pstatus-graphql-ktor/src/main/kotlin/gov/cdc/ocio/processingstatusapi/models/submission/Reference.kt [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+import com.expediagroup.graphql.generator.annotations.GraphQLDescription
+
+/**
+ * References within a report.
+  * @property type String?
+ * @property key  String?
+ * @property value  String?
+ */
+@GraphQLDescription("Contains Report references.")
+data class Reference(
```
Comment (mkrystof) [inline pstatus-graphql-ktor/src/main/kotlin/gov/cdc/ocio/processingstatusapi/models/submission/Reference.kt:12]: Where is this `Reference` class used?  I think it should be deleted.

File: pstatus-graphql-ktor/src/main/kotlin/gov/cdc/ocio/processingstatusapi/models/submission/StageInfo.kt [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+ * Message stageInfo within a Report.
+ **
+ * @property service String?
+ * @property stage String?
+ * @property version String?
+ * @property status String?
+ * @property startProcessingTime Date?
+ * @property endProcessingTime Date?
+ */
+@GraphQLDescription("StageInfo.")
```
Comment (mkrystof) [inline pstatus-graphql-ktor/src/main/kotlin/gov/cdc/ocio/processingstatusapi/models/submission/StageInfo.kt:26]: Please make the GraphQLDescription annotations more descriptive and useful.

**Review by mkrystof — CHANGES_REQUESTED** [review]

File: pstatus-graphql-ktor/src/main/kotlin/gov/cdc/ocio/processingstatusapi/models/submission/Reference.kt (hunk 4 above)
Comment (manu-govind) [reply pstatus-graphql-ktor/src/main/kotlin/gov/cdc/ocio/processingstatusapi/models/submission/Reference.kt]: removed

File: pstatus-graphql-ktor/src/main/kotlin/gov/cdc/ocio/processingstatusapi/models/submission/MessageMetadata.kt [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+import com.expediagroup.graphql.generator.annotations.GraphQLDescription
+
+/**
+ * MessageMetadata within a report.
+ * @property provenance  Provenance
+
+ */
+@GraphQLDescription("Report metadata containing the disposition type, message identifier, index, aggregation of whether Single or Batch and the filename")
+data class MessageMetadata(
```
Comment (mkrystof) [inline pstatus-graphql-ktor/src/main/kotlin/gov/cdc/ocio/processingstatusapi/models/submission/MessageMetadata.kt:11]: MessageMetadata should be this:
Screenshot 2024-08-01 at 9 48 33 AM

File: pstatus-graphql-ktor/src/main/kotlin/gov/cdc/ocio/processingstatusapi/models/submission/StageInfo.kt [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+ * @property endProcessingTime Date?
+ */
+@GraphQLDescription("Contains information about report service, action, version and whether it was a success or failure and the processing start and end times ")
+data class StageInfo(
+
+    @GraphQLDescription("Service")
+    var service : String? = null,
+
+    @GraphQLDescription("Stage name a.k.a action")
+    var stage: String? = null,
```
Comment (mkrystof) [inline pstatus-graphql-ktor/src/main/kotlin/gov/cdc/ocio/processingstatusapi/models/submission/StageInfo.kt:50]: This is a typo in the readme - this should be called `action`, not `stage`.

File: pstatus-graphql-ktor/src/main/kotlin/gov/cdc/ocio/processingstatusapi/models/submission/Issue.kt [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+/**
+ * Issues within a report.
+ *
+ * @property items Items?
+ */
+@GraphQLDescription("Contains Report issues.")
+data class Issue(
+
+    @GraphQLDescription("Items")
+    var items : Items? = null
```
Comment (mkrystof) [inline pstatus-graphql-ktor/src/main/kotlin/gov/cdc/ocio/processingstatusapi/models/submission/Issue.kt:14]: The `Issue` class should only contain `level` and `message`.  The `Items` and `Properties` classes aren't needed.

File: pstatus-graphql-ktor/src/main/kotlin/gov/cdc/ocio/processingstatusapi/models/submission/Provenance.kt [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+/**
+ * Provenance within a MessageMetadata.
+ **
+ * @property messageUUID String?
+ * @property messageHash String?
+ * @property aggregation String?
+ * @property messageIndex Int?`
+ */
+@GraphQLDescription("Contains Report metadata provenance.")
+data class Provenance(
```
Comment (mkrystof) [inline pstatus-graphql-ktor/src/main/kotlin/gov/cdc/ocio/processingstatusapi/models/submission/Provenance.kt:27]: Where did this come from?  Not mentioned in the readme.

**Review by mkrystof — CHANGES_REQUESTED** [review]

**Review by mkrystof — APPROVED** [review]

**Pull request merged**